src/requirements/requirement-graph-builder-from-user-data.ts

```diff
@@ -99,7 +106,113 @@ export default function buildRequirementFulfillmentGraphFromUserData(
+    /* keepCoursesWithoutDoubleCountingEliminationChoice */ false
+  ).requirementFulfillmentGraph;
+
+  /**
+   * The table below summerizes the type of all possible requirement-course edges before
+   * double-counting elimination, and where they will go in the new format.
+   *
+   * -------------------------------------------------------------------------------------------------
+   * | Requirement Type | edge exists after double-counting elim | where is it in the new format     |
+   * | ---------------- | -------------------------------------- | --------------------------------- |
+   * | selected         | True                                   | implicit (connected by default)   |
+   * | (no warning)     |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | selected         | True                                   | `acknowledgedCheckerWarningOptIn` |
+   * | (has warning)    |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | allow double     | True                                   | implicit (connected by default)   |
+   * | counting         |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | not connected    | False                                  | `optOut`                          |
+   * | (no warning)     |                                        |                                   |
+   * |                  |                                        |                                   |
+   * | not connected    | False                                  | implicit (unconnected by default) |
+   * | (has warning)    |                                        |                                   |
+   */
+  return function builder(course) {
+    const connectedRequirementsWithDoubleCountingAccounted = graphWithDoubleCountingAccounted.getConnectedRequirementsFromCourse(
+      course
+    );
+    const connectedRequirementsWithoutDoubleCountingAccounted = graphWithoutDoubleCountingAccounted.getConnectedRequirementsFromCourse(
+      course
+    );
+    /**
+     * complementary == All unconnected requirement after double counting elimination
+     *
+     * It's true that
+     * ```
+     * union(
+     *  selectableRequirementChoices[course],
+     *  requirements that allow double counting and the given course can be used to satisfy,
+     *  complementary,
+     * ) == all requirements that course can be used to satisfy
+     * ```
+     */
+    const complementary = new Set(connectedRequirementsWithoutDoubleCountingAccounted);
+    connectedRequirementsWithDoubleCountingAccounted.forEach(r => complementary.delete(r));
```

Comment:
Could you change the callback for `forEach` to be `complementary.delete` instead of `r => complementary.delete(r)`?

Reply:
That will actually crash the program due to JS `this` receiver unbinding. A simple example:

<img width="641" alt="Screen Shot 2021-11-03 at 11 52 19" src="https://user-images.githubusercontent.com/4290500/140095409-e54021d3-124a-4f8d-a191-56a850fa9470.png">